packages/flutter/test/painting/text_painter_test.dart

```diff
@@ -116,13 +116,15 @@ void main() {
     caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 18), ui.Rect.zero);
     expect(caretOffset.dx, 98); // 👧
     caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 19), ui.Rect.zero);
-    expect(caretOffset.dx, 98); // 🇺
+    expect(caretOffset.dx, 98); // 👏
     caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 20), ui.Rect.zero);
-    expect(caretOffset.dx, 112); // 🇺
+    expect(caretOffset.dx, 98); // 👏
```

Comment:
Change to a different type of emoji lgtm, but shouldn't we move the caret once the offset position is half-way through the hand emoji? (which is what the flag was doing) This behavior seems like it is not moving the caret at all until the end of line is reached/surpassed

Reply:
Though I am not totally clear what the expected behavior is with modifiers

Reply:
The positioning of the caret halfway through the flag was a bug.

In the test environment that uses the Ahem font, the "U" and "S" characters that encode the emoji were not being combined into one US flag grapheme.  So getOffsetForCaret(20) was yielding a position between the "U" and the "S".

The test environment does correctly handle the emoji with a modifier.  The framework will position the caret to the left of the last grapheme until reaching the end of the line.

Reply:
Ahh ok, thanks for the explanation. LGTM then!